pyccel/ast/core.py

```diff
@@ -1278,20 +1279,29 @@ def module(self):
         return self._module
 
 class Program(ScopedAstNode):
+    """
+    Represents a Program in the code.
 
-    """Represents a Program in the code. A block consists of the following inputs
+    A class representing a program in the code. A program is a set of statements
+    that are executed when the module is run directly. In Python these statements
+    are located in an `if __name__ == '__main__':` block.
 
     Parameters
     ----------
-    variables: list
-        list of the variables that appear in the block.
+    name : str
+        The name used to identify the program (this is used for printing in Fortran).
+
+    variables : list[Variable]
+        A list of the variables that appear in the program.
 
-    body: list
-        a list of statements
+    body : Iterable[PyccelAstNode]
+        An iterable object containing the statements in the body of the program.
 
-    imports: list, tuple
-        list of needed imports
+    imports : Iterable[Import]
+        An iterable object containing the imports used by the program.
```

Comment:
This is not strictly related to this PR, but it is worth addressing here nonetheless:

- Do we need `variables` to be a list, given that we only check that it is an iterable below?
- In the `__init__` we store `_imports = tuple(set(imports))` as a tuple, but we set `_variables = variables` and `_body = body`, which is potentially dangerous. I suggest to convert them to the proper type (probably a list).

Reply:
af892c8a96ed163fbfacbd5e2c16dfb315679e57